src/write.rs

```diff
@@ -915,27 +916,18 @@ impl<'parent, W: io::Write + io::Seek> SampleWriter16<'parent, W> {
         let s = sample.as_i16() as u16;
 
         // Write the sample in little endian to the buffer.
-        self.buffer[self.index as usize] = s as u8;
-        self.buffer[self.index as usize + 1] = (s >> 8) as u8;
+        self.buffer[self.index as usize].write(s as u8);
+        self.buffer[self.index as usize + 1].write((s >> 8) as u8);
 
         self.index += 2;
     }
 
-    #[cfg(target_arch = "x86_64")]
-    unsafe fn write_u16_le_unchecked(&mut self, value: u16) {
-        // x86_64 is little endian, so we do not need to shuffle bytes around;
-        // we can just store the 16-bit integer in the buffer directly.
-        let ptr = self.buffer.get_unchecked_mut(self.index as usize) as *mut u8 as *mut u16;
-        *ptr = value;
-    }
+    unsafe fn write_u16_unchecked(&mut self, value: u16) {
```

Comment:
```suggestion
    unsafe fn write_u16_le_unchecked(&mut self, value: u16) {
```
Can you preserve the `le` in the name? Without it, the representation of the `u16` in the output is ambiguous, the `le` clarifies that we write the bytes as little-endian.